src/utils/useEvents.ts

```diff
@@ -32,15 +37,15 @@ const useEvents = <T extends AllLeafletInstances | Map>(
       })
     }
     return () => {
-      if (eventsArray.length) {
+      if (mapInstance && eventsArray.length) {
```

Comment:
The `length` check can be removed as you can call `forEach` on an empty array.

```suggestion
    if (mapInstance) {
```